rotkehlchen/chain/ethereum/modules/eth2/beacon.py

```diff
@@ -234,14 +238,28 @@ def get_validator_data(
                 except DeserializationError:
                     log.error(f'Could not deserialize 0x01 withdrawal credentials for {entry}')
 
+            if queried_beaconchain:
+                if withdrawable_ts is not None and (exit_epoch := entry.get('exit_epoch', 0)) != 0:
+                    exited_ts = epoch_to_timestamp(exit_epoch)
+
+            elif withdrawable_ts is not None:  # query this index from beaconchain to see if exited
+                indices_mapping_to_query_beaconchain[entry[index_key]] = idx
```

Comment:
can't you change the order and check first `withdrawable_ts`?

```suggestion
            if withdrawable_ts is not None:
                if  queried_beaconchain and (exit_epoch := entry.get('exit_epoch', 0)) != 0:
                    exited_ts = epoch_to_timestamp(exit_epoch)
                else:  # query this index from beaconchain to see if exited
                    indices_mapping_to_query_beaconchain[entry[index_key]] = idx
```

unless I'm mistaken is the same condition with one check less